### ISSUE TITLE
silent should be true by default maybe?

### ISSUE THREAD

**Issue by Jakobud** [tanbrian/bookshelf-mass-assignment#1]
I'm new to Bookshelf and came across this plugin. It works very well. Honestly it should be apart of Bookshelf's core code imo. I came from Laravel's ORM which is how I can across a desire to protect my inputs on my models.

Anyways, so when using this plugin, you get an error thrown if a parameter is trying to be set that does not exist on the model. You can avoid this error if you pass the `{silent: true}` on the save method obviously.

I feel like this thing should be working silently by default. It shouldn't be throwing errors. The entire purpose of this plugin is to protect/guard your model from invalid or protected inputs from the user. So when would someone EVER actually desire an error to be thrown? You are always going to be protecting your model EVERY time it is saved. And since you don't want your application to crash when someone is using it (even if they are trying to maliciously pass an invalid attribute to your model), you are ALWAYS going to want to use `{silent: true}`. So why isn't it just that way by default?

The way Laravel's Eloquent ORM deals with this is that before a model is saved, it compares the mass assignment input to what is allowed (`fillable`) and what is `guarded`. It simply "throws out" any attributes that are `guarded` and any attributes that are not in the `fillable` array. Then it saves it. No errors thrown. It simply ignores what it needs to ignore and saves your model.

This is exactly how this plugin should be behaving. It shouldn't be throwing errors by default or else you simply will have a broken application unless you use `{silent: true}` EVERYWHERE.

Does this make sense? The only exception that I could think of is if someone wanted to log an error whenever an invalid or guarded attribute is tried to be saved on a model. But even in this case you would still want the model to save properly and silently to the user. I would think this would be a good opportunity to just have an event that is triggered so that the developer could log it behind the scenes if they wanted to.

Is this making sense?

**Comment by tanbrian** [comment]
Yep, that makes sense. I've admittedly been out of the JS world for a while and haven't been maintaining this package recently, but I'll take a look when I can. Thanks for the feedback!
